pkg/k8s/factory_functions.go

```diff
@@ -215,3 +216,126 @@ func EqualV1Namespace(ns1, ns2 *v1.Namespace) bool {
 	return ns1.Name == ns2.Name &&
 		comparator.MapStringEquals(ns1.GetLabels(), ns2.GetLabels())
 }
+
+func ConvertToNetworkPolicy(obj interface{}) interface{} {
+	netPol, ok := obj.(*networkingv1.NetworkPolicy)
+	if !ok {
+		return nil
+	}
+	return &types.NetworkPolicy{
+		NetworkPolicy: netPol,
+	}
+}
+
+func ConvertToK8sService(obj interface{}) interface{} {
+	service, ok := obj.(*v1.Service)
+	if !ok {
+		return nil
+	}
+	return &types.Service{
+		Service: service,
+	}
+}
+
+func ConvertToK8sEndpoints(obj interface{}) interface{} {
+	endpoints, ok := obj.(*v1.Endpoints)
+	if !ok {
+		return nil
+	}
+	return &types.Endpoints{
+		Endpoints: endpoints,
+	}
+}
+
+func ConvertToIngress(obj interface{}) interface{} {
+	ingress, ok := obj.(*v1beta1.Ingress)
+	if !ok {
+		return nil
+	}
+	ing := &types.Ingress{
+		Ingress: ingress,
+	}
+
+	*ingress = v1beta1.Ingress{}
+	return ing
+}
+
+func ConvertToCNPWithStatus(obj interface{}) interface{} {
+	cnp, ok := obj.(*cilium_v2.CiliumNetworkPolicy)
+	if !ok {
+		return nil
+	}
+	slimCNP := &types.SlimCNP{
```

Comment:
Can you explain what is being eliminated from the original CNP here?

Reply:
Nothing is.